modules/tinymce/src/core/main/ts/api/EditorOptions.ts

```diff
@@ -0,0 +1,284 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import { Arr, Fun, Obj, Optional, Strings, Type } from '@ephox/katamari';
+
+import Editor from './Editor';
+import { EditorOptions, NormalizedEditorOptions } from './OptionTypes';
+
+interface ProcessorSuccess<T> {
+  valid: true;
+  value: T;
+}
+
+interface ProcessorError {
+  valid: false;
+  message: string;
+}
+
+type SimpleProcessor = (value: unknown) => boolean;
+type Processor<T> = (value: unknown) => ProcessorSuccess<T> | ProcessorError;
+
+export interface BuiltInOptionTypeMap {
+  'string': string;
+  'number': number;
+  'boolean': boolean;
+  'string[]': string[];
+  'array': any[];
+  'function': Function;
+  'object': any;
+}
+
+export type BuiltInOptionType = keyof BuiltInOptionTypeMap;
+
+interface BaseOptionSpec {
+  immutable?: boolean;
+  deprecated?: boolean;
+  docsUrl?: string;
+}
+
+export interface BuiltInOptionSpec<K extends BuiltInOptionType> extends BaseOptionSpec {
+  processor: K;
+  default?: BuiltInOptionTypeMap[K];
+}
+
+export interface SimpleOptionSpec<T> extends BaseOptionSpec {
+  processor: SimpleProcessor;
+  default?: T;
+}
+
+export interface OptionSpec<T, U> extends BaseOptionSpec {
+  processor: Processor<U>;
+  default?: T;
+}
+
+/**
+ * TinyMCE Options API.
+ *
+ * @class tinymce.editor.Options
+ */
+
+export interface Options {
+  /**
+   * Register a new option that can be used within TinyMCE.
+   *
+   * @method register
+   * @param {String} name Name of an option.
+   * @param {OptionSpec} spec An option spec describing how to validate the option and other optional metadata.
```

Comment:
```suggestion
   * @param {OptionSpec} spec An option spec describing how to validate the option with other optional metadata.
```
little vague, could be read as "describing how to validate the option and how to validate the other optional metadata" although it that's what it does, you should keep "and" and add a "both" after describing?